daemon/cmd/daemon.go

```diff
@@ -188,15 +188,14 @@ func (d *Daemon) init() error {
 		}
 
 		if option.Config.SockopsEnable {
-			disableSockops := func(err error) {
-				option.Config.SockopsEnable = false
-				log.WithError(err).Warn("Disabled '--sockops-enable' due to missing BPF kernel support")
+			errSockops := func(err error) {
+				log.WithError(err).Error("BPF kernel support for 'sockops' missing")
```

Comment:
If we're failing here and we've properly checked for support in `initEnv()`, we're not failing because of missing kernel support. I think the error message needs to be updated to reflect this. Maybe something like `Failed to enable {Sockmsg,Sockmap}.`?

Reply:
fixed this